woocommerce/woocommerce-gateway-stripe
# Fix processPayment getting called when being paid by WooPayments on order pay page

Fixes #3041 

## Changes proposed in this Pull Request:
- Stripe script was getting enqueued when the gateway was disabled. Now we are bailing if gateway is diabled.
- Added a check in the `processPayment` function so that we don't try to process when the payment method is null.

## Testing instructions
1. Create a JN site and install the latest released versions of WooPayments and Stripe plugin.
2. Setup Stripe and WCPay in a JN site.
3. Disable Stripe as a gateway but leave the plugin active.
4. From `wp-admin` create an order with pending status. Then navigate to the `Customer payment page` (Pay for Order page) from the order edit page.
5. Try to pay for the order using the credit card/debit card option of WooPayments.
6. You will get the following error in console.
 `can't access property "elements", b[r] is undefined`
7. Now enabled the Stripe gateway, enabled UPE and card payment. 
8. Navigate to the Pay for Order page again and try to pay for the order using the credit card/debit card option of WooPayments.
9. You will get the same error.
 `can't access property "elements", b[r] is undefined`
10. Build this branch locally and upload the zip in your site.
11. Test both scenarios again (Stripe disabled and Stripe enabled with UPE) for order pay.
12. This time the payment should be successful.

### Additional testing
- Test payment using Stripe methods on the Pay for Order page when both WooPayments and Stripe is enabled as well as only Stripe is enabled.
- Test payment using Stripe methods on the checkout page when both WooPayments and Stripe is enabled as well as only Stripe is enabled.
- Test saving card from My Account page.
